### PR TITLE
PHP 8.0 Support

### LINKED ISSUE #8: PHP 8.0 support
### Feature Request

|    Q        |   A
|------------ | ------
| New Feature | yes

#### Summary

To be prepared for the december release of PHP 8.0, this repository has some additional TODOs to be tested against the new major version.

In order to make this repository compatible, one has to follow these steps:

- [ ] Modify `composer.json` to provide support for PHP 8.0 by adding the constraint `~8.0.0`
- [ ] Modify `composer.json` to drop support for PHP less than 7.3
- [ ] Modify `composer.json` to implement phpunit 9.3 which supports PHP 7.3+
- [ ] Modify `.travis.yml` to ignore platform requirements when installing composer dependencies (simply add `--ignore-platform-reqs` to `COMPOSER_ARGS` env variable)
- [ ] Modify `.travis.yml` to add PHP 8.0 to the matrix (*NOTE*: Do not allow failures as PHP 8.0 has a feature freeze since 2020-08-04!)
- [ ] Modify source code in case there are incompatibilities with PHP 8.0

### PR DESCRIPTION
**Pull request by Gounlaf**
This PR should check all tasks in #8 

Regards

### CODE REVIEW

**Comment by Gounlaf** [comment]
Have to rollback typehinting (BC Break)

**Review by andypost — APPROVED** [review]
Looks great and pass tests

**Review by froschdesign — CHANGES_REQUESTED** [review]
> Have to rollback typehinting (BC Break)

A rollback is correct here.

**Comment by Gounlaf** [comment]
> > Have to rollback typehinting (BC Break)
> 
> A rollback is correct here.

It was a reminder. I pushed another commit after that to do the rollback as required 🙂

**Comment by froschdesign** [comment]
@Gounlaf 
Take your time and many thanks for your help! 👍 👍 👍

**Comment by Gounlaf** [comment]
I've reverted changes related to typehinting, and squash.

**Review by boesing — APPROVED** [review]
LGTM.
I'd prefer having the `vendor/phpunit/phpunit/phpunit.xsd` instead for the schema ref but imho thats not a showstopper.

**Comment by Gounlaf** [comment]
> LGTM.
> I'd prefer having the `vendor/phpunit/phpunit/phpunit.xsd` instead for the schema ref but imho thats not a showstopper.

I can change it if you want

**Comment by andypost** [comment]
Is there release schedule for PHP 8 support? 
Drupal 9.1 (upcoming release) has it as dev-dependency and it's a blocker to declare PHP 8 support

File: phpunit.xml.dist [hunk 1]
```diff
@@ -1,17 +1,16 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:noNamespaceSchemaLocation="vendor/phpunit/phpunit/phpunit.xsd"
+         xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/9.3/phpunit.xsd"
```
Comment (froschdesign) [inline phpunit.xml.dist]: Please revert this and use the XML schema definition from the `vendor` directory of PHPUnit.

File: test/EscaperTest.php [hunk 2]
```diff
@@ -174,25 +174,23 @@ class EscaperTest extends TestCase
...
-     * @expectedException \Laminas\Escaper\Exception\InvalidArgumentException
-     * @expectedExceptionMessage Laminas\Escaper\Escaper constructor parameter must be a string, received integer
-     */
     public function testSettingEncodingToNonStringShouldThrowException()
     {
-        $escaper = new Escaper(1);
+        $this->expectExceptionMessage(
+            "Laminas\Escaper\Escaper constructor parameter must be a string, received integer"
+        );
+        $this->expectException(\Laminas\Escaper\Exception\InvalidArgumentException::class);
```
Comment (froschdesign) [inline test/EscaperTest.php]: Please import the class name.

File: test/EscaperTest.php [hunk 3]
```diff
@@ -174,25 +174,23 @@ class EscaperTest extends TestCase
...
+        new Escaper(1);
     }
 
-    /**
-     * @expectedException \Laminas\Escaper\Exception\InvalidArgumentException
-     */
     public function testSettingEncodingToEmptyStringShouldThrowException()
     {
+        $this->expectException(\Laminas\Escaper\Exception\InvalidArgumentException::class);
```
Comment (froschdesign) [inline test/EscaperTest.php]: The same here.

File: .travis.yml [hunk 4]
```diff
@@ -12,38 +12,21 @@ env:
...
       env:
         - DEPS=latest
-    - php: 7.3
+    - php: nightly
       env:
-        - DEPS=lowest
-    - php: 7.3
-      env:
-        - DEPS=latest
+        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```
Comment (froschdesign) [inline .travis.yml]: We use the option for dependencies also for nightly versions
```yaml
    - php: nightly
      env:
        - DEPS=lowest
        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
    - php: nightly
      env:
        - DEPS=latest
        - COMPOSER_ARGS="--no-interaction --ignore-platform-reqs"
```

**Review by froschdesign — CHANGES_REQUESTED** [review]

**Comment by Gounlaf** [comment]
@froschdesign I have commited changes according to your review.
I will squash once it's OK

**Comment by froschdesign** [comment]
@Gounlaf 
You are fast! 😃 👍

**Review by froschdesign — APPROVED** [review]

**Comment by Gounlaf** [comment]
> @Gounlaf
> You are fast! 😃 👍

:)

I can squash so?

**Review by andypost — APPROVED** [review]
Looks great 👍

**Comment by boesing** [comment]
Thanks, @Gounlaf!

**Pull request merged**